src/Microsoft.DotNet.Configurer/NuGetCachePrimer.cs

```diff
@@ -0,0 +1,115 @@
+        private readonly ICommandFactory _commandFactory;
+        private readonly IDirectory _directory;
+        private readonly IFile _file;
+        private readonly INuGetPackagesArchiver _nugetPackagesArchiver;
+        private readonly INuGetCacheSentinel _nuGetCacheSentinel;
+
+        public NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel)
+            : this(commandFactory,
+                nugetPackagesArchiver,
+                nuGetCacheSentinel,
+                FileSystemWrapper.Default.Directory,
+                FileSystemWrapper.Default.File)
+        {
+        }
+
+        internal NuGetCachePrimer(
+            ICommandFactory commandFactory,
+            INuGetPackagesArchiver nugetPackagesArchiver,
+            INuGetCacheSentinel nuGetCacheSentinel,
+            IDirectory directory,
+            IFile file)
+        {
+            _commandFactory = commandFactory;
+            _directory = directory;
+            _nugetPackagesArchiver = nugetPackagesArchiver;
+            _nuGetCacheSentinel = nuGetCacheSentinel;
+            _file = file;
+        }
+
+        public void PrimeCache()
+        {
+            if(SkipPrimingTheCache())
+            {
+                return;
+            }
+
+            var extractedPackagesArchiveDirectory = _nugetPackagesArchiver.ExtractArchive();
+
+            PrimeCacheUsingArchive(extractedPackagesArchiveDirectory);
+        }
+
+        private bool SkipPrimingTheCache()
+        {
+            return !_file.Exists(_nugetPackagesArchiver.NuGetPackagesArchive);
+        }
+
+        private void PrimeCacheUsingArchive(string extractedPackagesArchiveDirectory)
+        {
+            using (var temporaryDotnetNewDirectory = _directory.CreateTemporaryDirectory())
+            {
+                var workingDirectory = temporaryDotnetNewDirectory.DirectoryPath;
+                var createProjectSucceeded = CreateTemporaryProject(workingDirectory);
+
+                if (createProjectSucceeded)
+                {
+                    var restoreProjectSucceeded =
+                        RestoreTemporaryProject(extractedPackagesArchiveDirectory, workingDirectory);
+                    if (restoreProjectSucceeded)
+                    {
+                        _nuGetCacheSentinel.CreateIfNotExists();
+                    }
+                }
+            }
+        }
+
+        private bool CreateTemporaryProject(string workingDirectory)
+        {
+            return RunCommand("new", Enumerable.Empty<string>(), workingDirectory);
+        }
+
+        private bool RestoreTemporaryProject(string extractedPackagesArchiveDirectory, string workingDirectory)
+        {
+            return RunCommand(
+                "restore",
+                new[] {NUGET_SOURCE_PARAMETER, $"{extractedPackagesArchiveDirectory}"},
```

Comment:
So many other things are hard-coded inline, NUGET_SOURCE_PARAMETER feels out of place.  How about just inlining "--source" ?